exonum/src/events/noise/wrappers/sodium_wrapper/handshake.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::wrapper::NoiseWrapper;
+use crypto::{PublicKey, SecretKey};
+use events::{codec::MessagesCodec,
+             noise::{read, write, Handshake, HandshakeResult}};
+use futures::future::{done, Future};
+use std::io;
+use tokio_io::{codec::Framed, AsyncRead, AsyncWrite};
+
+#[derive(Debug, Clone)]
+/// Params needed to establish secured connection using Noise Protocol.
```

Comment:
If I don't miss anything, usually we place doc comments above `#[derive..]` constructions.